tests/test_acm/test_acm.py

```diff
@@ -0,0 +1,357 @@
+            {'Key': 'key2'},
+            {'Key': 'key3', 'Value': 'value3'},
+            {'Key': 'key4', 'Value': 'value4'},
+        ]
+    )
+
+    client.remove_tags_from_certificate(
+        CertificateArn=arn,
+        Tags=[
+            {'Key': 'key1', 'Value': 'value2'},  # Should not remove as doesnt match
+            {'Key': 'key2'},  # Single key removal
+            {'Key': 'key3', 'Value': 'value3'},  # Exact match removal
+            {'Key': 'key4'}  # Partial match removal
+        ]
+    )
+
+    resp = client.list_tags_for_certificate(CertificateArn=arn)
+    tags = {item['Key']: item.get('Value', '__NONE__') for item in resp['Tags']}
+
+    for key in ('key2', 'key3', 'key4'):
+        tags.should_not.contain(key)
+
+    tags.should.contain('key1')
+
+
+@mock_acm
+def test_remove_tags_from_invalid_certificate():
+    client = boto3.client('acm', region_name='eu-central-1')
+
+    try:
+        client.remove_tags_from_certificate(
+            CertificateArn=BAD_ARN,
+            Tags=[
+                {'Key': 'key1', 'Value': 'value1'},
+                {'Key': 'key2'},
+            ]
+        )
+    except ClientError as err:
+        err.response['Error']['Code'].should.equal('ResourceNotFoundException')
+    else:
+        raise RuntimeError('Should of raised ResourceNotFoundException')
+
+
+@mock_acm
+def test_resend_validation_email():
+    client = boto3.client('acm', region_name='eu-central-1')
+    arn = _import_cert(client)
+
+    client.resend_validation_email(
+        CertificateArn=arn,
+        Domain='*.moto.com',
+        ValidationDomain='NOTUSEDYET'
+    )
+    # Returns nothing, boto would raise Exceptions otherwise
+
+
+@mock_acm
+def test_resend_validation_email_invalid():
+    client = boto3.client('acm', region_name='eu-central-1')
+    arn = _import_cert(client)
+
+    try:
+        client.resend_validation_email(
+            CertificateArn=arn,
+            Domain='no-match.moto.com',
+            ValidationDomain='NOTUSEDYET'
+        )
+    except ClientError as err:
+        err.response['Error']['Code'].should.equal('InvalidDomainValidationOptionsException')
+    else:
+        raise RuntimeError('Should of raised InvalidDomainValidationOptionsException')
+
+    try:
+        client.resend_validation_email(
+            CertificateArn=BAD_ARN,
+            Domain='no-match.moto.com',
+            ValidationDomain='NOTUSEDYET'
+        )
+    except ClientError as err:
+        err.response['Error']['Code'].should.equal('ResourceNotFoundException')
+    else:
+        raise RuntimeError('Should of raised ResourceNotFoundException')
+
+
+@mock_acm
+def test_request_certificate():
+    client = boto3.client('acm', region_name='eu-central-1')
+
+    resp = client.request_certificate(
+        DomainName='google.com',
+        SubjectAlternativeNames=['google.com', 'www.google.com', 'mail.google.com'],
+    )
+    resp.should.contain('CertificateArn')
+
+
+# # Also tests the SAN code
```

Comment:
I normally don't like commented-out code but this seems like a great trade-off to me. Any future developer can take it from here if they notice the FreezeGun PR is merged

Reply:
Yeah agreed, but then this would be worth it. If that code gets merged then it would allow cutting time off many tests.

Reply:
FreezeGun PR has been merged already.    Bringing back these tests by  https://github.com/spulec/moto/pull/3372